django_tex/core.py

```diff
@@ -12,10 +12,13 @@
 
 def run_tex(source):
     with tempfile.TemporaryDirectory() as tempdir:
+        filename = os.path.join(tempdir, 'texput.tex')
+        with open(filename, 'x', encoding='utf-8') as f:
+            f.write(source)
         latex_interpreter = getattr(settings, 'LATEX_INTERPRETER', DEFAULT_INTERPRETER)
-        latex_command = [latex_interpreter, '-output-directory', tempdir]
-        process = Popen(latex_command, stdin=PIPE, stdout=PIPE, stderr=PIPE)
-        process.communicate(source.encode('utf-8'))
+        latex_command = [f'cd {tempdir}', '&&', latex_interpreter]
```

Comment:
you might want something like:

```python
command = [latex_interpreter, f'-output-directory={tempdir}', '-interaction=batchmode', 'texput.tex']
process = Popen(command, cwd=tempdir, stdout=PIPE, stderr=PIPE)
process.communicate()
```

The advantage here is that `shell=True` can allow arbitrary strings of commands to be run via the LATEX_INTERPRETER command whereas leaving it to `False` will only run one process, which is a bit safer. Plus you don't have to cd into anything or whatever.

Reply:
If `shell=False`, the test suite fails on travis-ci, since the latex binaries aren't found.

If I pass the complete path of the temporary tex file to `latexmk`, it works on travis (with `shell=True`), but fails on my local Windows machine, claiming that "Filename '...' contains character not allowed for TeX file". The path does not contain any whitespaces, so don't have any explanation for this error.

It seems, we should go with `shell=True` and cd'ing to the temporary directory.